test/tls13encryptiontest.c

```diff
@@ -0,0 +1,304 @@
+        "0000000000000002"
+    }
+};
+
+static int load_record(SSL3_RECORD *rec, size_t recnum, unsigned char **key,
+                       unsigned char *iv, size_t ivlen, unsigned char *seq)
+{
+    unsigned char *pt = NULL, *sq = NULL, *ivtmp = NULL;;
+    long ptlen;
+
+    *key = OPENSSL_hexstr2buf(refdata[recnum].key, NULL);
+    ivtmp = OPENSSL_hexstr2buf(refdata[recnum].iv, NULL);
+    sq = OPENSSL_hexstr2buf(refdata[recnum].seq, NULL);
+    pt = OPENSSL_hexstr2buf(refdata[recnum].plaintext, &ptlen);
+
+    if (*key == NULL || ivtmp == NULL || sq == NULL || pt == NULL)
+        goto err;
+
+    rec->data = rec->input = OPENSSL_malloc(ptlen + EVP_GCM_TLS_TAG_LEN);
+
+    if (rec->data == NULL)
+        goto err;
+
+    rec->length = ptlen;
+    memcpy(rec->data, pt, ptlen);
+    OPENSSL_free(pt);
+    memcpy(seq, sq, SEQ_NUM_SIZE);
+    OPENSSL_free(sq);
+    memcpy(iv, ivtmp, ivlen);
+    OPENSSL_free(ivtmp);
+
+    return 1;
+ err:
+    OPENSSL_free(*key);
+    *key = NULL;
+    OPENSSL_free(ivtmp);
+    OPENSSL_free(sq);
+    OPENSSL_free(pt);
+    return 0;
+}
+
+static int test_record(SSL3_RECORD *rec, size_t recnum, int enc)
+{
+    int ret = 0;
+    unsigned char *refd;
+    long refdatalenl;
+    size_t refdatalen;
+
+    if (enc)
+        refd = OPENSSL_hexstr2buf(refdata[recnum].ciphertext, &refdatalenl);
+    else
+        refd = OPENSSL_hexstr2buf(refdata[recnum].plaintext, &refdatalenl);
+
+    if (refd == NULL) {
+        fprintf(stderr, "Failed to get reference data for record %"OSSLzu
+                         "\n", recnum);
+        goto err;
+    }
+    refdatalen = (size_t)refdatalenl;
+
+    if (rec->length != refdatalen) {
+        fprintf(stderr, "Unexpected length for record %"OSSLzu"\n", recnum);
+        goto err;
+    }
+
+    if (memcmp(rec->data, refd, refdatalen) != 0) {
+        fprintf(stderr, "Data does not match for record %"OSSLzu"\n", recnum);
+        goto err;
+    }
+
+    ret = 1;
+
+ err:
+    OPENSSL_free(refd);
+    return ret;
+}
+static int test_tls13_encryption(void)
+{
+    SSL_CTX *ctx = NULL;
+    SSL *s = NULL;
+    SSL3_RECORD rec;
+    unsigned char *key = NULL, *iv = NULL, *seq = NULL;
+    const EVP_CIPHER *ciph = EVP_aes_128_gcm();
+    int ret = 0;
+    size_t ivlen, ctr;
+
+    rec.data = NULL;
+
+    ctx = SSL_CTX_new(TLS_method());
+    if (ctx == NULL) {
+        fprintf(stderr, "Failed creating SSL_CTX\n");
+        goto err;
+    }
+
+    s = SSL_new(ctx);
+    if (s == NULL) {
+        fprintf(stderr, "Failed creating SSL\n");
+        goto err;
+    }
+
+    s->enc_read_ctx = EVP_CIPHER_CTX_new();
+    s->enc_write_ctx = EVP_CIPHER_CTX_new();
+    if (s->enc_read_ctx == NULL || s->enc_write_ctx == NULL) {
+        fprintf(stderr, "Failed creating EVP_CIPHER_CTX\n");
+        goto err;
+    }
+
+    for (ctr = 0; ctr < OSSL_NELEM(refdata); ctr++) {
+        /*
+         * Load the record, set up the read/write sequences and load the key into
```

Comment:
If you're going to keep this multi-line comment, then the blank lines between the memcpy and the EVP_CIPHER setup should be removed.  Comments should go only for a single "paragraph" of code ...  Or split into three comments.